client.go

```diff
@@ -508,6 +804,36 @@ func (c *Client) Write(req *ua.WriteRequest) (*ua.WriteResponse, error) {
 	return res, err
 }
 
+// Republish executes a synchronous republish request.
+func (c *Client) Republish(req *ua.RepublishRequest) (*ua.RepublishResponse, error) {
+	var res *ua.RepublishResponse
+	err := c.sechan.SendRequest(req, c.Session().resp.AuthenticationToken, func(v interface{}) error {
+		return safeAssign(v, &res)
+	})
+	return res, err
+}
+
+// Publish executes a synchronous publish request.
+func (c *Client) Publish(acks []*ua.SubscriptionAcknowledgement) (*ua.PublishResponse, error) {
```

Comment:
I don't think it was a good idea to move `Publish` from `Subscription` to here. First, you changed the behavior: timeout in `Subscription.publish()` was intentionally set to a longer value than `c.cfg.RequestTimeout`.
Second, this method has an inconsistent with others signature. Other similar methods in `Client` accept corresponding requests as their argument, but this one accepts `[]*ua.SubscriptionAcknowledgement`.

Reply:
There must be a misunderstanding, In a previous message I tought you wanted to let the possibility to expose Republish in order for the client to implement their own reconnection. I made the assumption that the same goes for Publish.

![image](https://user-images.githubusercontent.com/25567035/72899119-a6d0af80-3d25-11ea-91cf-bf73ffe4d9ab.png)

Reply:
I understand where the confusion is coming from, but I really meant only Republish. I wouldn't even object to Publish being exposed in Client, but the special timeout for Publish as implemented in Subscription should stay there.

Just for simplicity's sake, I'd suggest that you only move Republish to Client and leave Publish as is for now.